javascript/javascript1/week3/notes-app.js

```diff
@@ -0,0 +1,62 @@
+const notes = [];
+
+// Function for saving note
+function saveNote(content, id) {
+    notes.push({ "content": content, "id": id });  // push object to notes array 
+    return notes;
+}
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+saveNote("Wash dishes", 3);
+
+console.log(notes);
+
+
+
+// Function for getting a specific note
+function getNote(id) {
+
+    if (!id || typeof id !== "number") {
+        return "An error string";
+    } else {
+        for (let i = 0; i < notes.length; i++) {
+            if (notes[i].id === id) {
+                return notes[i];
+            }
+        }
+        return "The id is not in any object";
+    }
+}
+
+const firstNote = getNote(2);
+console.log(firstNote); // {content: 'Pick up groceries', id: 1}
+
+
+
+// Function for logging out notes
+function logOutNotesFormatted() {
+    for (let i in notes) {
+        console.log(`The note with id: ${notes[i].id}, has the following note text: ${notes[i].content}`);
+    }
+}
+
+logOutNotesFormatted();
+
+
+
+// Additional function for deleting specific note using id
+function deleteObjectWithId(arr, id) {
+    const objIdIndex = arr.findIndex((obj) => obj.id === id);
+
+    for (let i = 0; i < arr.length; i++) {
```

Comment:
Let's think about two cases. 
if we assume that `id` is unique (e.g. there can not be more than one notes with the same id` then there's no need in the loop, it would be enough to 
```javascript
const objIdIndex = arr.findIndex((obj) => obj.id === id);
arr.splice(objIdIndex, 1);
```

But if we think that `id` is not unique, then `arr.splice(objIdIndex, 1);` in the loop should be `arr.splice(i, 1);`